core/domain/caching_services.py

```diff
@@ -0,0 +1,188 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service functions to set and retrieve data from the memory cache."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import collection_domain
+from core.domain import exp_domain
+from core.domain import skill_domain
+from core.domain import story_domain
+from core.domain import topic_domain
+from core.platform import models
+import python_utils
+
+memory_cache_services = models.Registry.import_cache_services()
+DESERIALIZATION_FUNCTIONS = {
+    'collection': collection_domain.Collection.deserialize,
+    'exploration': exp_domain.Exploration.deserialize,
+    'skill': skill_domain.Skill.deserialize,
+    'story': story_domain.Story.deserialize,
+    'topic': topic_domain.Topic.deserialize,
+    'default': lambda x: x
+}
+
+
+def flush_memory_cache():
+    """Flushes the memory cache by wiping all of the data."""
+    memory_cache_services.flush_cache()
+
+
+def _get_deserialization_function_for_namespace(namespace):
+    """Get the deserialization function associated with the namespace specified.
+
+    Args:
+        namespace: str. The namespace that differentiates the objects.
+
+    Returns:
+        Method. The deserialization function associated with that particular
+        namespace.
+    """
+    if namespace not in DESERIALIZATION_FUNCTIONS:
```

Comment:
All namespaces should be in DESERIALIZATION_FUNCTIONS.keys(). If the passed-in namespace is not, throw an error.